src/components/Media/Media.tsx

```diff
@@ -35,40 +29,27 @@ const Media: React.FC<MediaProps> = ({ items, activeItem }) => {
   return (
     <MediaWrapper>
       {items.map((item: object, key: number) => {
-        // 1. get canvas
-        const canvas: CanvasNormalized = vault.fromRef(item);
-        // 2. get annotationPage
-        const annotationPage: AnnotationPageNormalized = vault.fromRef(
-          canvas.items[0],
-        );
-        // 3.get annotations
-        const annotations: Array<AnnotationNormalized> = vault.allFromRef(
-          annotationPage.items,
+        // this probably needs to be written in a .filter()
+        const canvasEntity: object = getCanvasByCriteria(
+          vault,
+          item,
+          motivation,
+          paintingType,
         );
 
-        // 4.determine if W3C annotation is motivation painting
-        for (const annotation of annotations) {
-          if (annotation.motivation.includes(annotationMotivation)) {
-            // 5.get contentResource (the annotation body)
-            const contentResource: ContentResource = vault.fromRef(
-              annotation.body[0],
-            );
-            // 6. render media item if annotation target matches original
-            //    canvas ID and contentResource type is Sound or Video
-            if (
-              annotation.target === item.id &&
-              contentResourceType.includes(contentResource.type)
-            )
-              return (
-                <MediaItem
-                  key={key}
-                  canvas={canvas}
-                  active={true}
-                  handleChange={handleChange}
-                />
-              );
-          }
-        }
+        let isActive = false;
+        if (activeCanvas === item.id) isActive = true;
+
+        if (canvasEntity !== undefined)
+          return (
+            <MediaItem
+              active={isActive}
+              canvasEntity={canvasEntity}
+              thumbnail={getThumbnail(vault, canvasEntity, 200, null)}
+              key={key}
```

Comment:
This `key` value is coming from the `map` function's `index` value?   If so, and we have a unique id or piece of data for each `item`, lets try and use that instead of using array indexes as keys.